operator/Makefile

```diff
@@ -159,58 +164,65 @@ everything: build bundle ## Build everything (local binary, operator image, bund
+
+.PHONY: controller-gen
+controller-gen: $(CONTROLLER_GEN) ## Download controller-gen locally if necessary.
+$(CONTROLLER_GEN): $(LOCALBIN)
+	test -s $(LOCALBIN)/controller-gen || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_TOOLS_VERSION)
 
-OPERATOR_SDK_VERSION = 1.20.1
-OPERATOR_SDK = $(PROJECT_DIR)/bin/operator-sdk-$(OPERATOR_SDK_VERSION)
+.PHONY: envtest
+envtest: $(ENVTEST) ## Download envtest-setup locally if necessary.
+$(ENVTEST): $(LOCALBIN)
+	test -s $(LOCALBIN)/setup-envtest || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
+
+OPERATOR_SDK_VERSION = 1.24.1
+OPERATOR_SDK = $(LOCALBIN)/operator-sdk-$(OPERATOR_SDK_VERSION)
 .PHONY: operator-sdk
 operator-sdk: ## Download operator-sdk necessary for scaffolding and bundling.
 # See https://sdk.operatorframework.io/docs/installation/#install-from-github-release
-	@ARCH=$$(case $$(uname -m) in (x86_64) echo amd64 ;; (aarch64) echo arm64 ;; (*) uname -m ;; esac) ;\
+	@ARCH=$$(case $$(uname -m) in (x86_64) echo -n amd64 ;; (aarch64) echo -n arm64 ;; (*) echo -n $$(uname -m) ;; esac) ;\
 	OPERATOR_SDK_URL=https://github.com/operator-framework/operator-sdk/releases/download/v$(OPERATOR_SDK_VERSION)/operator-sdk_$(OS)_$${ARCH} ;\
 	$(GET_GITHUB_RELEASE_FN); \
 	get_github_release $(OPERATOR_SDK) $${OPERATOR_SDK_URL}
 
-KUTTL_VERSION = 0.15.0
-KUTTL_UPSTREAM = kudobuilder
-KUTTL ?= $(PROJECT_DIR)/bin/kubectl-kuttl-$(KUTTL_VERSION)
+KUTTL_VERSION = 0.14.0-kubeconfig
+KUTTL_UPSTREAM = porridge
+KUTTL ?= $(LOCALBIN)/kubectl-kuttl-$(KUTTL_VERSION)
 .PHONY: kuttl
 kuttl: ## Download kuttl.
-	@ARCH=$$(case $$(uname -m) in (aarch64) echo armv6 ;; (*) uname -m ;; esac) ;\
+	@ARCH=$$(case $$(uname -m) in (aarch64) echo -n armv6 ;; (*) echo -n $$(uname -m) ;; esac) ;\
 	KUTTL_URL=https://github.com/$(KUTTL_UPSTREAM)/kuttl/releases/download/v$(KUTTL_VERSION)/kubectl-kuttl_$(KUTTL_VERSION)_$(OS)_$${ARCH} ;\
 	$(GET_GITHUB_RELEASE_FN); \
 	get_github_release $(KUTTL) $${KUTTL_URL}
 
-ENVTEST = $(PROJECT_DIR)/bin/setup-envtest
-.PHONY: envtest
-envtest: ## Download envtest-setup locally if necessary.
-	$(call go-get-tool,$(ENVTEST),sigs.k8s.io/controller-runtime/tools/setup-envtest@latest)
-
-# go-get-tool will 'go get -d' any package $2 and install it to $(PROJECT_DIR)/bin unless $1 already exists.
-define go-get-tool
-$(SILENT)[ -x $(1) ] || { \
-set -e ;\
-TMP_DIR=$$(mktemp -d) ;\
-cd $$TMP_DIR ;\
-go mod init tmp ;\
-echo "Downloading $(2)" ;\
-go get -d $(2) ;\
-go build -o $(1) $(firstword $(subst @, ,$(2))) ;\
-rm -rf $$TMP_DIR ;\
+GET_GITHUB_RELEASE_FN = get_github_release() { \
+	[ -x $${1} ] || { \
+		set -euxo pipefail ;\
+		mkdir -p bin ;\
+		attempts=5 ;\
+		for i in $$(seq $$attempts); do \
+			curl --fail --location --output $${1} $${2} && break ;\
+			[ $$i -eq $$attempts ] && exit 1; \
+		done ;\
+		[[ "$$(uname -s)" != "Darwin" ]] || xattr -c $${1} ;\
+		chmod +x $${1} ;\
+	} \
```

Comment:
Looks suspiciously similar to [operator/hack/get-github-release.sh](https://github.com/stackrox/stackrox/blob/master/operator/hack/get-github-release.sh). Could it be reused?

Reply:
Looks like @kurlov missed the fact that this moved to `make/github.mk` ?

Reply:
Sorry, I made a mistake during rebasing